airbyte-integrations/connectors/source-mysql-v2/src/main/kotlin/io/airbyte/integrations/source/mysql/cdc/MySqlDebeziumOperations.kt

```diff
@@ -0,0 +1,308 @@
+            }
+        }
+        return Jsons.objectNode().apply { set<JsonNode>(STATE, stateNode) }
+    }
+
+    val databaseName: String = jdbcSourceConfiguration.namespaces.first()
+    val serverID: Int = random.nextInt(MIN_SERVER_ID..MAX_SERVER_ID)
+
+    val commonProperties: Map<String, String> by lazy {
+        val tunnelSession: TunnelSession = jdbcConnectionFactory.ensureTunnelSession()
+        DebeziumPropertiesBuilder()
+            .withDefault()
+            .withConnector(MySqlConnector::class.java)
+            .withDebeziumName(databaseName)
+            .withHeartbeats(cdcSourceConfiguration.debeziumHeartbeatInterval)
+            // This to make sure that binary data represented as a base64-encoded String.
+            // https://debezium.io/documentation/reference/2.2/connectors/mysql.html#mysql-property-binary-handling-mode
+            .with("binary.handling.mode", "base64")
+            // https://debezium.io/documentation/reference/2.2/connectors/mysql.html#mysql-property-snapshot-mode
+            .with("snapshot.mode", "when_needed")
+            // https://debezium.io/documentation/reference/2.2/connectors/mysql.html#mysql-property-snapshot-locking-mode
+            // This is to make sure other database clients are allowed to write to a table while
+            // Airbyte is taking a snapshot. There is a risk involved that if any database client
+            // makes a schema change then the sync might break
+            .with("snapshot.locking.mode", "none")
+            // https://debezium.io/documentation/reference/2.2/connectors/mysql.html#mysql-property-include-schema-changes
+            .with("include.schema.changes", "false")
+            .withDatabase(jdbcSourceConfiguration.jdbcProperties)
+            .withDatabase("hostname", tunnelSession.address.hostName)
+            .withDatabase("port", tunnelSession.address.port.toString())
+            .withDatabase("dbname", databaseName)
+            .withDatabase("server.id", serverID.toString())
+            .withDatabase("include.list", databaseName)
+            .withOffset()
+            .withSchemaHistory()
+            // TODO: add missing properties, like MySQL converters, etc. Do a full audit.
+            .buildMap()
+    }
+
+    val syntheticProperties: Map<String, String> by lazy {
+        DebeziumPropertiesBuilder()
+            .with(commonProperties)
+            // https://debezium.io/documentation/reference/2.2/connectors/mysql.html#mysql-property-snapshot-mode
+            // We use the recovery property cause using this mode will instruct Debezium to
+            // construct the db schema history. Note that we used to use schema_only_recovery mode
+            // instead, but this mode has been deprecated.
+            .with("snapshot.mode", "recovery")
+            .withStreams(listOf())
+            .buildMap()
+    }
+
+    companion object {
+        const val MIN_SERVER_ID = 5400
+        const val MAX_SERVER_ID = 6400
```

Comment:
What are those IDs for?

Reply:
Honestly, I don't know why those values in particular were picked. I carried this over from the source-mysql connector. They're for https://debezium.io/documentation/reference/stable/connectors/mysql.html#mysql-property-database-server-id
Perhaps the randomization isn't even necessary in this test. I'll see what I can do.

Reply:
Added a comment.